service/src/main/java/bio/terra/workspace/app/controller/ControlledAzureResourceApiController.java

```diff
@@ -791,6 +801,28 @@ public ResponseEntity<ApiAzureManagedIdentityResource> getAzureManagedIdentity(
         fetchCreateControlledKubernetesNamespaceResult(jobId), HttpStatus.OK);
   }
 
+  /**
+   * Initial implementations of createAzureKubernetesNamespace and createAzureDatabase required ids
+   * of the resources to be passed in. But names are easier to use and port nicely during clone
+   * operations. So currently the api supports both names and ids. But we convert to names before
+   * storing the resource.
+   *
+   * @param workspaceId
+   * @param maybeUuid a uuid or a name
+   * @return if maybeUuid is a uuid and exists in the workspace, return the name of the resource
+   *     with that uuid. Otherwise, return maybeUuid.
+   */
+  private String maybeLookupName(UUID workspaceId, String maybeUuid) {
```

Comment:
We are trying to locate resource when 'id' is passed, but In case 'maybeUuid' represents name of a resource should we validate that such resource exists?

Reply:
the validation happens later on in one of the stairway steps